crates/noirc_frontend/src/parser/parser.rs

```diff
@@ -403,9 +437,8 @@ fn function_declaration_parameters() -> impl NoirParser<Vec<(Ident, UnresolvedTy
     let typ = parse_type().recover_via(parameter_recovery());
     let typ = just(Token::Colon).ignore_then(typ);
 
-    let parameter = ident().recover_via(parameter_name_recovery()).then(typ);
-
-    let parameter = parameter.or(self_parameter().validate(|param, span, emit| {
+    let full_parameter = ident().recover_via(parameter_name_recovery()).then(typ);
+    let self_parameter = self_parameter().validate(|param, span, emit| {
```

Comment:
This changes are simply to remove the shadowing that happened before. In my view, these changes simply make the code easier to understand for the reader. As the problem you outlined, `check function declarations don't contain any patterns`, I will be happy to address it in separate issue/PR.

Reply:
My mistake, I was thinking you changed it so we only `validate`d on the self parameter but after looking at the original code more closely, only checking on the `self` parameter is correct anyway since we only parse an identifier for the full parameter. I was thinking of `function_parameters` where we parse a full pattern.